### ISSUE TITLE
Order of precedence of CSS different from vite-plugin-vue

### ISSUE THREAD

**Issue by Tahul** [nuxt/framework#7847]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-hkou23?file=app.vue,components%2FBlock.vue,main.ts,package.json

Run Vite env: `npm run vite`
Run Nuxt env: `npm run nuxt`

Apps runs same components, one with Vite, the other with Nuxt.

### Describe the bug

It seems like when running Nuxt, order of precedence of CSS only gets respected in SSR, but at the moment the app mounts, every scoped CSS declaration will get overwritten by parent.

That results in a flash in styling (first page render showing same CSS as Vite envs, then it blinks to another state).

I suspect this is coming from order of precedence of CSS that is respected in Vite env, but not in Nuxt.

I am not sure it is a bug or some of the "caveats" of scoped CSS, I just noticed the difference while working with `<style>` and thought it would be safer to report.

**Comment by Tahul** [comment]
https://stackblitz.com/edit/nuxt-starter-tgwwz6?file=pages%2Findex.vue,components%2FBlock.vue

I made the same reproduction in Nuxt 2, and the behavior seem to be the same between Nuxt 2 and 3, so I guess this is more of a limitations from style scoped?

**Comment by danielroe** [comment]
Yes, this is an issue of CSS specificity and (therefore) ordering, and in general I would advise against relying on the _order_ in which equally specific rules in two different components appear.

It's also a little bit of an edge case. [Put the blocks in another component](https://stackblitz.com/edit/nuxt-starter-y8rfdr?file=app.vue,components%2FBlock.vue,main.ts,components%2FBlocks.vue) and there is no longer a conflict as the scoped data value is not applied any more.

But worth investigating. The strange thing is that it is the _vite_ build that orders the CSS differently in each case - that is, just looking at the two different `.css` files, they are ordered differently.
